src/logger/BaseLogger.go

```diff
@@ -0,0 +1,144 @@
+package logger
+
+//this doesn't resolve, don't know why
+//not super important right now
+// import (
+// 	"net/url"
+// )
+
+/**
+ * Initialize enabled logger using default url.
+ */
+func NewBaseLoggerAgent(_agent string) *BaseLogger {
+	return NewBaseLogger(_agent, "insert default url here", true, nil)
+}
+
+/**
+ * Initialize enabled/disabled logger using default url.
+ */
+func NewBaseLoggerAgentEnabled(_agent string, _enabled bool) *BaseLogger {
+	return NewBaseLogger(_agent, "insert default url here", _enabled, nil)
+}
+
+/**
+ * Initialize enabled logger using url.
+ */
+func NewBaseLoggerAgentUrl(_agent string, _url string) *BaseLogger {
+	return NewBaseLogger(_agent, _url, true, nil)
+}
+
+/**
+ * Initialize enabled logger using queue.
+ */
+func NewBaseLoggerAgentQueue(_agent string, _queue []string) *BaseLogger {
+	return NewBaseLogger(_agent, "insert default url here", true, nil)
+}
+
+/**
+* Initialize enabled/disabled logger using queue.
+ */
+func NewBaseLoggerAgentQueueEnabled(_agent string, _queue []string, _enabled bool) *BaseLogger {
+	return NewBaseLogger(_agent, "insert default url here", _enabled, nil)
```

Comment:
should pass queue to the constructor and the baselogger's URL needs to be set to nil

Reply:
Enableable isn't set here, just the "enabled" property. I can't say whether the structure of these constructors should be changed yet, I'm just trying to replicate the functionality of the java version of the class. I did miss using the _queue argument, good catch!